newIDE/app/src/AssetStore/AssetStoreNavigator.js

```diff
@@ -164,7 +164,12 @@ export const useShopNavigation = (): NavigationState => {
             previousHistory.previousPages[
               previousHistory.previousPages.length - 1
             ];
-          if (!isSearchResultPage(currentPage)) {
+          if (isSearchResultPage(currentPage)) {
+            // The search has changed, show the best results.
+            currentPage.pageBreakIndex = 0;
+            currentPage.scrollPosition = 0;
+            return previousHistory;
```

Comment:
There is a risk that React is confused because you return the same object, which React can interpret as "nothing changed so the component won't be re-rendered" (or components will skip re-rendering a prop that is the same as before).

Instead, do:
```suggestion
    // The search has changed, show the best results.
    const updatedCurrentPage = {
      ...currentPage,
      pageBreakIndex: 0,
      scrollPosition: 0,
    };
    
            return {...previousHistory, previousPages: [
              ...previousHistory.previousPages.slice(0, -1), // All pages except the last one
              updatedCurrentPage,
            ]};
```

Reply:
I wonder if doing this will trigger necessary UI update.

Actually, the page index is a state so the page change is not to trigger an update of the UI, it's only to update the history that will be used when we go back to this page.

Reply:
I would say that if it trigger an unnecessary UI update, it's a problem of this UI (that should maybe use pure component or something else), but not a problem of this state, that does it job of NOT mutating an object but replacing it if it changes. 

If the re-render is too long, it's not the state problem, but your components that should have useMemo/React.memo or other things.

Reply:
From my experience, you always get bitten because "I was just mutating something" which then 3 months later turns into "oops so the filter/history panel did not refresh"